libcodechecker/libhandlers/parse.py

```diff
@@ -278,20 +320,28 @@ def main(args):
             files = [os.path.join(input_path, file_name) for file_name
                      in file_names]
 
+        file_change = False
         for file_path in files:
-            report_stats = parse(file_path,
-                                 context,
-                                 metadata_dict,
-                                 suppress_handler,
-                                 skip_handler,
-                                 'print_steps' in args)
+            report_stats, f_change = parse(file_path,
+                                           context,
+                                           metadata_dict,
+                                           suppress_handler,
+                                           skip_handler,
+                                           'print_steps' in args)
+            if not file_change:
+                file_change = f_change
 
             severity_stats.update(Counter(report_stats.get('severity',
                                           {})))
             file_stats.update(Counter(report_stats.get('files', {})))
             report_count.update(Counter(report_stats.get('reports', {})))
 
         print("\n----==== Summary ====----")
+        if file_change:
```

Comment:
For my eyes it would be better to put this warning after summary. So the output would be this:
```bash
----==== Summary ====----
------------------------
Filename  | Report count
------------------------
main2.cpp |            1
------------------------
-----------------------
Severity | Report count
-----------------------
HIGH     |            1
-----------------------
----=================----
Total number of reports: 1
----=================----

[2018-02-09 12:14] - Some source file contents are changed since the last analysis. Multiple reports were not shown and skipped from the statistics. Please analyze your project again to update the reports!
```
Note: also place a new line before the warning (`print("")`).